services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataSchedulerActor.scala

```diff
@@ -111,8 +111,17 @@ class ArchiveMetadataSchedulerActor(archiveMetadataConfig: ArchiveMetadataConfig
     countWorkflowsLeftToArchiveThatEndedOnOrBeforeThresholdTimestamp(
       TerminalWorkflowStatuses,
       currentTimestampMinusDelay
-    ).map(workflowsToArchive => sendGauge(workflowsToArchiveMetricPath, workflowsToArchive.longValue(), ServicesPrefix))
-    ()
+    ).onComplete({
+      case Success(workflowsToArchive) =>
+        sendGauge(workflowsToArchiveMetricPath, workflowsToArchive.longValue(), ServicesPrefix)
+        // schedule next workflows left to archive query after interval
+        context.system.scheduler.scheduleOnce(archiveMetadataConfig.instrumentationInterval)(workflowsLeftToArchiveMetric())
+      case Failure(exception) =>
+        log.error(exception, s"Something went wrong while fetching number of workflows left to archive. " +
+          s"Scheduling next poll in ${archiveMetadataConfig.instrumentationInterval}.")
+        // schedule next workflows left to archive query after interval
+        context.system.scheduler.scheduleOnce(archiveMetadataConfig.instrumentationInterval)(workflowsLeftToArchiveMetric())
```

Comment:
TOL do we need to cancel this schedule when the actor gets stopped? What happens when you try to shut down Cromwell and this is running?

Reply:
If the actor dies, since it is `scheduleOnce`, the scheduling would stop no? Same with shutting down Cromwell, it would be stopped. Right?

Reply:
Ok, sounds good to me!